### PR TITLE
Write some basic routes tests

### LINKED ISSUE #184: Off by one error with blog posts: extra page exists
the fix should be in `app/routes/client.py`

screen shot 2015-04-16 at 11 49 18 am

### LINKED ISSUE #200: Write tests that ensure that all pages 200

### PR DESCRIPTION
**Pull request by danrschlosser**
Review: @eunicekokor @RaymondXu @natebrennand 

Closes #200, adding some basic routes tests

Also closes #184

### CODE REVIEW

File: test/base.py [hunk 1]
```diff
@@ -53,10 +62,6 @@ def setUpClass(self):  # noqa
         from app import app
         self.app = app
 
-    @classmethod
```
Comment (eunicekokor) [inline test/base.py]: why is this gone and does it need to be implemented somewhere (it wasn't previously)?

File: app/routes/blog.py [hunk 2]
```diff
@@ -57,13 +57,15 @@ def blog_archive(index):
...
     """
-    index = int(index)
-
     if index <= 0:
         return redirect(url_for('.index'))
 
     blog_posts = BlogPost.objects(published=True).order_by('-date_published')
-    if len(blog_posts) < 10 * (index + 1):
+
+    if len(blog_posts) <= 10 * index:
```
Comment (natebrennand) [inline app/routes/blog.py]: i think there should be comments on what these are doing

File: test/test_routes.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from base import TestingTemplate
+
+
+class TestRoutes(TestingTemplate):
+    """Test the basic behavior of routes in eventum."""
+
+    OK = (200, 201)
+    REDIRECT = (301, 302)
+
+    CLIENT_ROUTE_EXPECTATIONS = [
```
Comment (natebrennand) [inline test/test_routes.py]: yay table based tests :100:

File: test/test_routes.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        ('/events', OK),
+        ('/events/15', REDIRECT),
+        ('/resources', OK),
+        ('/labs', OK),
+        ('/blog', OK),
+        ('/blog/15', REDIRECT),
+        ('/contact', OK),
+    ]
+
+    EVENTUM_ROUTE_EXPECTATIONS = [
```
Comment (natebrennand) [inline test/test_routes.py]: Should we also assert that these redirect/404 when requested w/o a role?

File: test/base.py (hunk 1 above)
Comment (danrschlosser) [inline test/base.py]: This function gets called after the tests run.   This file is kinda messy as I look through it now... (I wrote it a *while* ago).   I'll clean things up, probably adding this back to do some `drop_collection()`s.

File: test/test_routes.py (hunk 4 above)
Comment (danrschlosser) [inline test/test_routes.py]: So right now tests are being run with google authentication off, so there isn't a great way to test that.  I'm not sure how to reconcile the need for a client_secrets file to run google auth and not wanting to publish client secrets to travis.

**Comment by natebrennand** [comment]
@danrschlosser, anything blocking this?

**Comment by danrschlosser** [comment]
Nope

**Comment by natebrennand** [comment]
:shipit:

**Pull request merged**